src/components/pages/MainPage.js

```diff
@@ -2,31 +2,35 @@ import React, {useState} from "react";
 import {Button, Spinner} from "react-mdl";
 import {Pages} from "../NavigationComponent";
 import {AuthContext} from "../../utils/Auth";
-import {createEnvironment} from "../../utils/api";
+import apiCall from "../../utils/api";
 
 const MainPage = ({nav}) => {
     const [isLoading, setIsLoading] = useState(false);
     const [isMfaSetup, setIsMfaSetup] = useState(false);
     const authContext = React.useContext(AuthContext);
 
     React.useEffect(() => {
-        async function checkMfaSetup() {
+        async function userSetup() {
+            const setup = await apiCall(authContext, "verify-user")
+            if (!setup) return nav.go(Pages.USER_NOT_SET_UP)}
+
 
+        async function checkMfaSetup() {
             const user = await authContext.getCurrentUser()
             if (!authContext.isFederated(user) && user.preferredMFA !== 'SOFTWARE_TOKEN_MFA') {
                 return nav.go(Pages.SETUP_MFA);
             }
 
             if (isMfaSetup === false) setIsMfaSetup(true);
         }
-
         checkMfaSetup();
+        userSetup();
```

Comment:
Can we not get a race condition here? If checkMFASetup and userSetup are both async function calls, couldn't we hit the checkMfaSetup trigger first and take us to the MFA setup? However, I guess this only applies for users we have setup ourselves since federated users don't have MFA.